File: odin-project-exercises/javascript-exercises/calculator/calculator.js
function add(a, b) {
  let sum = a + b;
  return sum;
}

function subtract(a, b) {
  let dif = a - b;
  return dif;
}

function sum(array) {
  let i;
  let sum = 0;
  for (i = 0; i < array.length; i++) sum += array[i];
  return sum;
}

function multiply(array) {
  product = array.reduce((a, b) => a * b);
  return product;
}

function power(a, b) {
  return a ** b;
}

function factorial(number) {
  let fact = 1;
  for (let i = number; i > 0; i--) {
    fact *= i;
  }
  return fact;
}

module.exports = {
  add,
  subtract,
  sum,
  multiply,
  power,
  factorial
};

File: odin-project-exercises/ruby/calculator.rb
#write your code here
def add a, b
    a + b
end

def subtract a, b
    a - b
end

def sum array
    sum = 0
    array.each {|x| sum += x}
    sum
end

def multiply *nums
    prod = 1
    nums.each { |n| prod *= n }
    prod
end

def power a, b
    a ** b
end

def factorial a
    prod = 1
    a.downto(1) { |n| prod *= n }
    prod
end